dbt/adapters/default/impl.py

```diff
@@ -131,42 +151,53 @@ def get_result_from_cursor(cls, cursor):
 
     @classmethod
     def drop(cls, profile, schema, relation, relation_type, model_name=None):
-        if relation_type == 'view':
-            return cls.drop_view(profile, schema, relation, model_name)
-        elif relation_type == 'table':
-            return cls.drop_table(profile, schema, relation, model_name)
-        else:
-            raise RuntimeError(
-                "Invalid relation_type '{}'"
-                .format(relation_type))
+        # add deprecation warning
+        identifier = relation
+        relation = cls.Relation.create(
+            schema=schema,
+            identifier=identifier,
+            type=relation_type)
+
+        return cls.drop_relation(profile, relation, model_name)
 
     @classmethod
-    def drop_relation(cls, profile, schema, rel_name, rel_type, model_name):
-        relation = cls.quote_schema_and_table(profile, schema, rel_name)
-        sql = 'drop {} if exists {} cascade'.format(rel_type, relation)
+    def drop_relation(cls, profile, relation, model_name=None):
+        sql = 'drop {} if exists {} cascade'.format(relation.type, relation)
```

Comment:
is `relation.type` required? If not: can we add an exception if `relation.type` is None here? I imagine that will be a common failure mode here